src/ansys/aedt/core/visualization/post/post_3dlayout.py

```diff
@@ -88,68 +161,23 @@ def compute_power_by_layer(self, layers=None, nets=None, solution=None):
         dict
             Power by layer.
         """
-        layers, nets, solution = self._check_inputs(layers, nets, solution)
-        if layers is None or nets is None or solution is None:
-            self._app.logger.error("Check inputs.")
-            return False
         power_by_layers = {}
-        solution_type = ["DC Fields"] if settings.aedt_version < "2025.1" else ["DCIR Fields"]
-        for layer in layers:
-            thickness = self._app.modeler.edb.stackup[layer].thickness
-            operations = []
-            idx = 0
-            for net in nets:
-                try:
-                    get_ids = self._app.odesign.GetGeometryIdsForNetLayerCombination(net, layer, solution)
-                except Exception:  # pragma no cover
-                    get_ids = []
-                if not get_ids:
-                    continue
-                assignment = f"{layer}_{net}"
-                operations.extend(
-                    [
-                        "Fundamental_Quantity('P')",
-                        f"EnterSurface('{assignment}')",
-                        "Operation('SurfaceValue')",
-                        "Operation('Integrate')",
-                    ]
-                )
-                idx += 1
-                if idx > 1:
-                    operations.append("Operation('+')")
-            operations.extend([f"Scalar_Constant({thickness})", "Operation('*')"])
-            if idx == 0:
-                continue
-            my_expression = {
-                "name": f"Power_{layer}",
-                "description": "Power Density",
-                "design_type": ["HFSS 3D Layout Design"],
-                "fields_type": solution_type,
-                "solution_type": "",
-                "primary_sweep": "",
-                "assignment": "",
-                "assignment_type": ["Surface"],
-                "operations": operations,
-                "report": ["Data Table", "Rectangular Plot"],
-            }
-            if self._app.post.fields_calculator.is_expression_defined(my_expression["name"]):
-                self._app.post.fields_calculator.delete_expression(my_expression["name"])
-            self._app.post.fields_calculator.add_expression(my_expression, "")
-            power_by_layers[layer] = self._app.post.fields_calculator.evaluate(
-                my_expression["name"], solution, intrinsics={}
-            )
+        power_loss = self._compute_power_loss(solution)
+        for i in power_loss:
+            layer_name = i["layer"]
+            loss = i["loss"]
+            if layer_name not in power_by_layers:
+                power_by_layers[layer_name] = loss
+            else:
+                power_by_layers[layer_name] += loss
         return power_by_layers
 
     @pyaedt_function_handler()
-    def compute_power_by_nets(self, nets=None, layers=None, solution=None):
+    def compute_power_by_net(self, solution=None):
```

Comment:
Same as above. I would also leave the filter solution (with None as default) to be able to compute for the entire board or only for the selected layers/nets